### ISSUE TITLE
Does Sophia works with activation-checkpointing?

### ISSUE THREAD

**Issue by ahmdtaha** [Liuhong99/Sophia#50]
Has anyone tried Sophia with activation-checkpointing? 
I am able to use sophia in a vanilla training setup, but once I add [checkpointing](https://github.com/prigoyal/pytorch_memonger/blob/master/tutorial/Checkpointing_for_PyTorch_models.ipynb),  torch.autograd.grad start throwing error about missing grad_fn.

**Comment by Liuhong99** [comment]
I think it should work with gradient checkpointing, although I did not try myself. The calculation of the gradient used in the approximation of the Hessian should be the same as the gradient used for parameter update, so it should work out-of-the-box with PyTorch gradient checkpointing function.
